mtools/test/test_util_logfile.py

```diff
@@ -114,3 +114,90 @@ def test_hostname_port(self):
         print(logfile2.hostname)
         assert logfile2.hostname == 'jimoleary.local'
         assert logfile2.port == '27017'
+
+    def test_shard_info(self):
+        """LogFile: test if sharding info is detected (MongoDB 3.4+) """
```

Comment:
If this should work in 3.4+, I'd use 3.4 as the baseline (or 3.6 since 3.4 is EOL)